### ISSUE TITLE
More false positives after unnecessary_lambda_linter() refactor

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2249]
After #2231 and #2247, I decided to see if any other issues may have crept in in the `unnecessary_lambda_linter()` refactor (#2186).

Lo and behold there seem to be a lot. I'm hoping the following covers everything:

```r
lint(linters = unnecessary_lambda_linter(), '{
    lapply(x, function(e) sprintf("%o", e))
    lapply(y, function(e) paste(strlpad(e, "0", width)))
}')
```

Slightly stylized from some code seen in base/R/version.R. This is about as minimal as I can make it for now.

**Comment by MichaelChirico** [comment]
OK, after #2250, I don't see any more obvious false positives for `unnecessary_lambda_linter()`.
